### PR TITLE
Buy Books

### LINKED ISSUE #178: Buy books as a user on Mobile
It should be made very clear in the report that this is a simulation of an actual purchase and not a puchase per say.

### PR DESCRIPTION
**Pull request by ZhakalenDk**
This also includes a fix for the `BookController.PurchaseBookAsync` always returning `DuplicateSpecification`
This closes #178

`[Unauthorized]`
image

`[Authorized - Owned]`
image

`[Authorized - Not Owned]`
image

`[Authorized - Purchased]`
image

### CODE REVIEW

File: PubHub/PubHub.API/Controllers/BooksController.cs [hunk 1]
```diff
@@ -161,7 +161,7 @@ public async Task<IResult> GetBookAsync(Guid id, [FromHeader] string appId)
         }
 
         [HttpPost("{id}/purchase")]
```
Comment (Sjimooon) [inline PubHub/PubHub.API/Controllers/BooksController.cs:163]: Did we want to rename this to 'purchases' or should we just leave it?

Comment (ZhakalenDk) [reply PubHub/PubHub.API/Controllers/BooksController.cs:163]: I csn rename it quick if that's desired 🤔🤷‍♂️

Comment (Sjimooon) [reply PubHub/PubHub.API/Controllers/BooksController.cs:163]: Personally I don't really care. But if the standard says no "actions" in the URL(?) 🤔

Comment (ZhakalenDk) [reply PubHub/PubHub.API/Controllers/BooksController.cs:163]: Yea. But we also broke the standard on users with `/suspend`

Comment (Sjimooon) [reply PubHub/PubHub.API/Controllers/BooksController.cs:163]: Yeah, lets just move on then 🫡

**Review by Sjimooon — APPROVED** [review]

**Pull request merged**